src/messages/messages.ts

```diff
@@ -99,3 +103,45 @@ export const createReactionMessage = (fromId: string, emoji: string, inReplyTo:
   fromId,
   inReplyTo,
 });
+
+/**
+ * hash() takes a DSNPMessage and returns a standard hash for use in signatures
+ * as defined in the [Message Signatures](https://github.com/LibertyDSNP/spec/blob/main/pages/Messages/Signatures.md)
+ * specification.
+ *
+ * @param   message The message to hash
+ * @returns         A Uint8Array of the resulting hash
+ */
+export const hash = (message: DSNPMessage): Uint8Array => {
+  const sortedMessage = sortObject((message as unknown) as Record<string, unknown>);
+  let serialization = "";
+
+  for (const key in sortedMessage) {
+    serialization = `${serialization}${key}${sortedMessage[key]}`;
+  }
+
+  return new Uint8Array(keccak256.digest(serialization));
+};
+
+/**
+ * sign() takes a private key and DSNP message and returns a Uint8Array
+ * signature for the message.
+ *
+ * @param privateKey The private key to use for signing
+ * @param message    The DSNP message to sign
+ * @returns          The message signature as a Uint8Array
+ */
+export const sign = (privateKey: Uint8Array, message: DSNPMessage): Uint8Array =>
+  secp256k1.ecdsaSign(hash(message), privateKey).signature;
+
```

Comment:
This feels like a wallet problem, not an SDK problem. I know we are skirting around that issue. Perhaps we need both?